assets/src/dashboard/app/router/routerProvider.js

```diff
@@ -19,7 +19,6 @@
  */
 import { useRef, useMemo, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { parse } from 'query-string';
```

Comment:
@BrittanyIRL Can you explain what is happening here? When I called `parse` from query-string I got an empty object every time. Do we know this was working?

Reply:
When looking at an individual template on the "Explore Templates" page, the URL will look like `wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/templates-gallery/template-detail?id=3&isLocal=false`

`parse()` turns the `?id=3&isLocal=false` part (`history.current.location.search`) into `{id: "3", isLocal: "false"}`

Reply:
I don't understand what is happening here. Will ask advice from @BrittanyIRL

Reply:
Pascal's right about what it's for. To expand on it more. Take the following: 
I am looking at the detail view of a template. The URL ends with: `/templates-gallery/template-detail?id=3&isLocal=false` 

This is grabbed from history location search of the router (looks like this: `?id=3&isLocal=false`) and is parsed with `parse` to become `{id: "3", isLocal: "false"}` which is stored as `queryParams`. 

`queryParams` is part of the Router's context (`state.queryParams`) and is important to setting what the view is in dashboard/app as well as what template is getting looked at in the detail view. 

So, it is doing something. Removing this will break the template view. 😂  

If you want to remove it we will need to parse the search still.

You can see the queryParam update if you log it like so: 

```
  useEffect(() => {
    console.log(queryParams);
  }, [queryParams]);
```

and go to the templates view and click on a template to go to detail view. 

hope this helps, let me know if you have more questions about it.